src/components/Profile/components/SchedulePanel/index.jsx

```diff
@@ -34,7 +44,34 @@ const GordonSchedulePanel = (props) => {
   const [reloadCall, setReloadCall] = useState(false);
   const [editDescriptionOpen, setEditDescriptionOpen] = useState(false);
   const [scheduleControlInfo, setScheduleControlInfo] = useState(null);
-  // Remove the line: let scheduleControlInfo = null;
+  const [sessions, setSessions] = useState([]);
+  const [eventInfo, setEventInfo] = useState([]);
+  const [currentAcademicSession, setCurrentAcademicSession] = useState('');
+
+  const [selectedSession, setSelectedSession] = useState('');
+  const isOnline = useNetworkStatus();
+
+  const sessionFromURL = new URLSearchParams(location.search).get('session');
+
+  useEffect(() => {
+    const loadPage = async () => {
+      setSessions(await sessionService.getAll());
+
+      if (sessionFromURL) {
+        setSelectedSession(sessionService.encodeSessionCode(sessionFromURL));
+      } else {
+        const { SessionCode: currentSessionCode } = await sessionService.getCurrent();
+        setCurrentAcademicSession(currentSessionCode);
+      }
+    };
+    loadPage();
+  }, [sessionFromURL]);
+
+  const handleSelectSession = async (value) => {
+    setSelectedSession(value);
+    value = sessionService.decodeSessionCode(value);
```

Comment:
```suggestion
    value = sessionService.decodeSessionCode(value);
```

Does something happen here by reference? You've already `setSelectedSession` so what is going on here?

Reply:
I remember previously when @mla04762 and I copy `handleSelectSession`from`involvementsAll``index.jsx`, and we replaced `navigate` with `reloadHandler()` so the schedule panel would auto change the schedule by term selected.

Reply:
This line code doesn't affect anything with schedule result, so I think we should delete it.